### PR TITLE
Gas miner refactor

### PR DESCRIPTION
**Pull request by adacovsk**
Kammer wanted to see the actual code differences, so here it is. I might as well stick it in as a real PR

## What this does
- refactor gas miners

## Why it's good
- I did the work already so if anyone wants to modify it in the future, it will be much easier for them

Tested: 
image

### CODE REVIEW

File: code/game/machinery/atmoalter/gas_mine.dm [hunk 1]
```diff
@@ -111,138 +127,48 @@
...
-	if(ticker)
-		air_contents.copy_from(gas_giant.GM)
-
-
-/obj/machinery/atmospherics/miner/mixed_nitrogen
-	name = "\improper Mixed Gas Miner"
-	desc = "Pumping nitrogen, carbon dioxide, and plasma."
-	overlay_color = "#FF80BD"
-
-/obj/machinery/atmospherics/miner/mixed_nitrogen/AddAir()
```
Comment (ShiftyRail) [inline code/game/machinery/atmoalter/gas_mine.dm:214]: Why are you removing those ?

File: code/game/machinery/atmoalter/gas_mine.dm [hunk 2]
```diff
@@ -111,138 +127,60 @@
...
-	name = "\improper Mixed Gas Miner"
-	desc = "Pumping oxygen and nitrous oxide."
-	overlay_color = "#7EA7E0"
-
-/obj/machinery/atmospherics/miner/mixed_oxygen/AddAir()
-  var/rate = AirRate()
-  air_contents.adjust_multi(GAS_OXYGEN, 0.5*rate,
-  GAS_SLEEPING, 0.5*rate)
-
-/obj/machinery/atmospherics/miner/gas_sink
```
Comment (ShiftyRail) [inline code/game/machinery/atmoalter/gas_mine.dm:230]: Why are you removing the gas sink ?

**Comment by bathosbathtime** [comment]
The mixed gas miners I made have been readded but you are still removing the gas sink miner, whatever that thing is.

**Comment by bathosbathtime** [comment]
Further, even if you managed to not remove things for your refactor you yourself admit that you made this solely to own kammerjunk in an argument and the idea that this makes it easier to code with the gas miners is something that I, someone who has done exactly that, find hard to believe.
discussion in the code channel

**Pull request closed without merging**